homeassistant/components/nad/media_player.py

```diff
@@ -200,12 +201,16 @@ def update(self):
         self._volume = self.calc_volume(self._nad_receiver.main_volume("?"))
```

Comment:
I want us to check for None already here before we call `self.calc_volume`. It's better to be strict with the types that functions accept and check types before calling functions.

Reply:
Yes, I understand what you mean, I think it makes the code uglier, but here you go :)